### DISCUSSION TITLE
About writing files

### DISCUSSION THREAD

**Discussion by LiuJiao0408** [Parcels-code/Parcels#1313] [Q&A]
Hello! I want to ask a question about writing files!

I am running an experiment, which has been completed, but I closed the program for other reasons.

The problem now is that the experiment has been completed and the results have not been written into the file (currently it is still an npy file). 

I don't want to rerun the program. It takes too long.

How can I rewrite the file.

**Comment by erikvansebille** [comment]
Dear @LiuJiao0408, thanks for your question. If you get temporary .npy files as output, it means that you are running an older version of Parcels. Since [version 2.4.0](https://github.com/OceanParcels/parcels/releases/tag/v2.4.0), output is directly saved to `.zarr` format which you can read immediately (even during the simulation) so that you don't have this problem anymore. I would therefore strongly recommend you update to the latest version

In the meantime, older versions of Parcels had a command line tool `parcels_convert_npydir_to_netcdf`wjhich you could use to convert an `out-*` directory to a netcdf file. You could try that tool? However, that tool had some memory limitations (see also the extensive discussion at https://github.com/OceanParcels/parcels/issues/1091), which is one of the reasons we moved to zarr-output in late 2022.
